src/ServiceControl.Audit/Auditing/AuditIngestionComponent.cs

```diff
@@ -43,6 +57,14 @@ AuditIngestionCustomCheck.State ingestionState
             watchdog = new Watchdog(ingestion.EnsureStarted, ingestion.EnsureStopped, ingestionState.ReportError,
                 ingestionState.Clear, settings.TimeToRestartAuditIngestionAfterFailure, log, "failed message ingestion");
 
+            channel = Channel.CreateBounded<MessageContext>(new BoundedChannelOptions(settings.MaximumConcurrencyLevel * 2)
```

Comment:
Is MaxConcurrencyLevel equivalent of how many in-flight messages are allowed by the semaphore? Wouldn't it then be impossible for the channel to fill with any more than the MaxConcurrencyLevel of messages?

Either way some comments on this line about those assumptions would be really good.

Reply:
I verified again and setting it to `MaxConcurrencyLevel` is good enough. I pushed a commit that changes it and does also specify the full mode enum even though it uses the default value. I've also added a few comments to the loop. 

In case you are interested to see how it behaves here is a sample

https://github.com/danielmarbach/RawAndChannel